### ISSUE TITLE
Ubuntu > Joplin Server > Android Sync - Android Note - No Image

### ISSUE THREAD

**Issue by blackfrank74** [laurent22/joplin#8525]
## Environment

Laptop
Joplin version:  Joplin 2.11.11 (prod, linux)
Platform: Linux 
OS specifics:  Ubuntu 22.04.2 LTS 64 bit

Joplin Server
Joplin version:  Joplin Server v2.12.1, copyright © 2021-2023 JOPLIN.
Platform: Asustor NAS   
OS specifics:  ADM4.2.2.RI61

Android phone
Joplin version:  2.11.32
Platform: Samsung Galaxy S10+   
OS specifics:  Android 12

Android tablet
Joplin version:  2.11.32
Platform: Samsung Galaxy Tab S7
OS specifics:  Android 13

--------------------------------------------------------------------------------------

## Steps to reproduce

here is an example good note on my Ubuntu install
![ubuntu note](https://github.com/laurent22/joplin/assets/34759786/d54629ae-060a-4b80-9136-23d07776c9e5)
And the note seems to have successfully synced to the Joplin Server.
![ubuntu sync status](https://github.com/laurent22/joplin/assets/34759786/a7ba35d8-45f7-4022-8a06-eaa13a232e93)

here is the same note on my phone
![andr phone  note](https://github.com/laurent22/joplin/assets/34759786/40638d31-8d1d-4f67-ba84-46f604a706d2)
and here is the sync status
![andr phone sync status](https://github.com/laurent22/joplin/assets/34759786/d2c1caa8-4d75-4b7c-bccd-1d87640bf21c)
Note the image displays on Ubuntu but not in the phone 

here is the same note on my tablet
![android tab note](https://github.com/laurent22/joplin/assets/34759786/20d8b6f5-d08c-429f-b028-f6f0ec25066a)
and here is the sync status
![android tab  sync status](https://github.com/laurent22/joplin/assets/34759786/fc9d13d8-835c-42e1-8fff-1902c1477d4d)
Note the image displays on Ubuntu but not in the tablet

This inconsistency one of many examples that I have noted.

1. per above + when I sync on the Android devices, no change to the note pic. 

--------------------------------------------------------------------------------------

## Describe what you expected to happen

I expect the pics will sync onto the Android devices

--------------------------------------------------------------------------------------

## Logfile

![android tab log](https://github.com/laurent22/joplin/assets/34759786/cd832e29-7681-4e98-9f83-da62b9080f4c)
Best I can see, all errors pertain to duplicate alarms

[android tab syncReport-1689934510541.zip](https://github.com/laurent22/joplin/files/12122628/android.tab.syncReport-1689934510541.zip)

**Comment by blackfrank74** [comment]
Attachment download behavior = "Always" on all Ubuntu and Android devices

**Comment by personalizedrefrigerator** [comment]
Is the date/time correct on all of the devices (including the server)?

If an image is attached from Android, does the image sync to the other devices?

**Comment by blackfrank74** [comment]
Hello, thanks for responding.

The time on the devices are set to auto sync to internet servers and align to within 1-2 minutes.

Here is a test note I created on the phone
![note on phone](https://github.com/laurent22/joplin/assets/34759786/5cbb5fc2-2346-40e9-9f20-28fbb2b6a7db)
And this is how came through on via Joplin Server sync on my Ubuntu device
![note on ubuntu](https://github.com/laurent22/joplin/assets/34759786/b0066f42-a6c6-413b-978a-31382697c7ed)

The sync problem seems to be bi-directional. 

On the Ubuntu device -
I've tried clicking 
- on this text string "Screenshot_20230721_192100_Joplin.png"
- on the down arrow image
(just in case, to see if it made a difference) - in both cases nothing happened.

Finally, despite several sync attempts I don't see the note created on the phone coming to the tablet at all.

**Comment by blackfrank74** [comment]
On the Ubuntu device -
I've tried clicking 
- on this text string "![Screenshot_20230721_192100_Joplin.png](:/59c4f4f914de40f9b434b23afa5498f8)"
- on the down arrow image
(just in case, to see if it made a difference) - in both cases nothing happened.

**Comment by blackfrank74** [comment]
Just checking in if this is on anyone's radar at all?

Laptop > Joplin server shows 9k notes and 15.7k resources synced. (screen print above) - I consider this the source of truth
In contrast -
Joplin server > phone shows 9k notes and 1.3k resources synced. (screen print above) - ie 14.4k resources did not sync
Joplin server > tablet shows 9k notes and 10k resources synced. (screen print above) - ie 5.7k resources did not sync

This resource mismatch is surely why the phone (and tablet) doesn't show all the graphics.
The android devices downstream from the Joplin server have been syncing manually/automatically all week with no change.

I need a way to trigger a full rescan/resync of the items so that any missing can sync over.
I've previously tried to do this by uninstalling the Joplin app from both phone and tablet, and resyncing via Joplin server but I never get everything over...  
Before that, I tried syncing via Dropbox but had extremely slow writing of records onto the android devices (apparently a known Android issue)
So I'm running out of options here.

The last (worst option) is to  to manually populate the Joplin folders on the Android devices, but I cannot see where the target Joplin folders are located on the Android devices. Regardless, I'd like to use a Joplin native sync solution rather than aa 3rd party app hack to sync.

**Comment by personalizedrefrigerator** [comment]
Have you tried the "delete remote data and re-upload" option in the desktop app's advanced sync settings? (Be sure you have an up-to-date backup before doing this!)

**Comment by blackfrank74** [comment]
I will try delete and reupload of the data on the Joplin server shortly.

The laptop > Joplin server sync status (if I interpret it correctly?) seem to indicate that right now this hop of the sync is ok. 
![ubuntu sync status](https://github.com/laurent22/joplin/assets/34759786/fc693dd4-0326-4496-8f21-a2c9d2fee6bb)

But - I think I see your reasoning - if the server file date/time stamps change then maybe the missing data will sync into the android devices.

**Comment by blackfrank74** [comment]
FYI There was no delete option, I am running the "Re-upload local data to sync target" option

**Comment by blackfrank74** [comment]
FYI There was no delete option, I am running the "Re-upload local data to sync target" option

**Comment by blackfrank74** [comment]
I ran the "Re-upload local data to sync target" option to refresh the data on the Joplin server.
And then clicked "synchronise" on Android mobile device 1, later on Android mobile device 2.
In both cases, the sync finished in 2-3 seconds only, ie no data transferred from the Joplin server to the devices.

Next I deleted the Joplin app from both Android mobile devices.
Re-installed it on mobile device A.
Installed the "screen alive" app from the google store to force the device to stay awake. (my Samsung S10 phone/ S7 tablet  cannot be set to stay awake for longer than 30 minutes)
And let the sync run overnight, without touching the device. 
After approx 9-10 hours the sync was complete, and the item counts on laptop (source of truth) = that on the mobile device A... so happy days
 
I then followed the same process for mobile device B.
Again, the item counts on laptop (source of truth) = that on the mobile device B... so happy days

Given this now works, and I can see attachments and images I will now close this issue.

Thanks for the help.
